File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    result = 0
    
    while storey:
        res = storey % 10
        
        if 5 < res:
            result += (10 - res)
            storey += 10
        elif res <= 4:
            result += res
        else:
            if (storey // 10) % 10 > 4:
                storey += 10
            result += res
        storey //= 10
    
    return result
            
            

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    let result = 0;
    
    while(storey !== 0){
        const res = storey % 10;
        if(5 < res){
            result += (10 - res);
            storey += 10;
        } else if(res < 5){
            result += res;
        } else {
            if(Math.floor(storey / 10) % 10 > 4) storey += 10
            result += res
        }
        storey = Math.floor(storey / 10);
    }
    
    return result;
}